core/components/ui/input/input.tsx

```diff
@@ -1,76 +1,50 @@
-import { cva } from 'class-variance-authority';
 import { AlertCircle, Check } from 'lucide-react';
-import { ComponentPropsWithRef, createContext, ElementRef, forwardRef, useContext } from 'react';
+import { ComponentPropsWithRef, ElementRef, forwardRef, ReactNode } from 'react';
 
 import { cn } from '~/lib/utils';
 
-const inputVariants = cva(
-  'peer focus-visible:ring-primary/20 w-full border-2 border-gray-200 py-2.5 px-4 text-base placeholder:text-gray-500 focus-visible:border-primary focus-visible:outline-none focus-visible:ring-4 hover:border-primary disabled:bg-gray-100 disabled:hover:border-gray-200',
-  {
-    variants: {
-      variant: {
-        success:
-          'pe-12 border-success-secondary focus-visible:border-success-secondary focus-visible:ring-success-secondary/20 disabled:border-gray-200 hover:border-success',
-        error:
-          'pe-12 border-error-secondary focus-visible:border-error-secondary focus-visible:ring-error-secondary/20 disabled:border-gray-200 hover:border-error',
-      },
-    },
-  },
-);
-
-type VariantTypes = 'success' | 'error';
-
-export interface InputProps extends ComponentPropsWithRef<'input'> {
-  variant?: VariantTypes;
+interface InputProps extends ComponentPropsWithRef<'input'> {
+  variant?: 'success' | 'error';
+  icon?: ReactNode;
 }
 
-const InputContext = createContext<{ variant?: VariantTypes }>({ variant: undefined });
-
-const InputIcon = forwardRef<ElementRef<'span'>, ComponentPropsWithRef<'span'>>(
-  ({ className, children, ...props }, ref) => {
-    const { variant } = useContext(InputContext);
-
-    return (
-      <span
-        aria-hidden="true"
+const Input = forwardRef<ElementRef<'input'>, InputProps>(
+  ({ className, variant, children, icon, type = 'text', ...props }, ref) => (
+    <div className={cn('relative', className)}>
+      <input
         className={cn(
-          'pointer-events-none absolute end-4 top-0 flex h-full items-center peer-disabled:text-gray-200',
-          variant === 'success' && 'text-success-secondary',
-          variant === 'error' && 'text-error-secondary',
+          'peer w-full border-2 border-gray-200 px-4 py-2.5 text-base placeholder:text-gray-500 hover:border-primary focus-visible:border-primary focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-primary/20 disabled:bg-gray-100 disabled:hover:border-gray-200',
+          variant === 'success' &&
+            'border-success-secondary pe-12 hover:border-success focus-visible:border-success-secondary focus-visible:ring-success-secondary/20 disabled:border-gray-200',
+          variant === 'error' &&
+            'border-error-secondary pe-12 hover:border-error focus-visible:border-error-secondary focus-visible:ring-error-secondary/20 disabled:border-gray-200',
           className,
```

Comment:
Looks like `className` already on the outer `<div>`

Reply:
Good catch, fixed.